[Koh Tee Sing Newton] iP

NewtonKoh opened the pull request:
# Jelly

> "You are the Jelly to my Peanut Butter" -Someone Probably ([source](https://google.com/))

Jelly is a chatbot who can help you manage tasks, events, and deadlines! He is:

- Fun and Encouraging
- Full of Energy
- ~~Easy~~ SUPER EASY to use

All you need to do is:

1. download it.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you. :+1:

And it is **FREE**

Features:

- [x] Manage tasks.
- [ ] ChatGPT support for conversations. (coming soon)
- [ ] Reminders. (coming in the future)

If you want to learn how to code, Here's the `main` method to print "Hi Jelly" to your console:

```java
public class Jelly {
    public static void main(String[] args) {
        System.out.println("Hi Jelly");
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

NewtonKoh committed:
Rename, Greet, Exit

NewtonKoh committed:
Echo

NewtonKoh committed:
Add, List

NewtonKoh committed:
Mark as Done

NewtonKoh committed:
Todos, Events, Deadlines

NewtonKoh committed:
Automated Text UI Testing

NewtonKoh committed:
Handle Errors

NewtonKoh committed:
Delete

NewtonKoh committed:
Use Enums

Not applicable

NewtonKoh committed:
Save

NewtonKoh committed:
Dates and Times

BryanL2303 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+
+    @Override
+    public String toString(){
+        if(isParsed){
```
There should be a space between 'if' and the open bracket '('.

In almost all places across all the classes you seem to have missed the space before the '(' when they are necessary. Such as while and for loops and if/else statements.

BryanL2303 commented on src/main/java/Deadline.java:
In every line where the class or methods are defined there is no space before '{'.
This also seems to be the case for the other classes.

BryanL2303 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+    @Override
+    public String toString(){
+        if(isParsed){
+
```
I don't think the coding standard mentions anything about leaving blank lines after the declaration of a method or class, but is there a reason why you do this? The toString method above does not have the blank line for example.

In all the other classes I also see many blank lines in random spots... Not sure if they are placed there on purpose but I can't really understand the logic of having those spaces at those spots.

BryanL2303 commented on src/main/java/Jelly.java:
```diff
@@ -0,0 +1,183 @@
+        String argument = "";
+
+        if(lines.length > 1) {
+        argument = message.substring(command.length() + 1);
```
Might have missed an identation here.

BryanL2303 commented on src/main/java/Jelly.java:
```diff
@@ -0,0 +1,183 @@
+
+        System.out.println(line);
+
+        switch(command){
```
Refering to the coding standard for switch commands the identations are different from what we might be used to. The lines of 'case' statements should be on the same identation level as the 'switch' statement itself. Only the items within the cases needs to be further indented

BryanL2303 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+
+        String string = "";
+
+        for(Integer i=1;i<=tasks.size();++i){
```
Generally we would always leave a space between operators.
I think this line should look like "Integer i = 1; i <= tasks.size(); ++i)"

NewtonKoh committed:
Use More OOP

NewtonKoh committed:
Organize into Packages

NewtonKoh committed:
Merge branch 'add-gradle-support'

NewtonKoh committed:
Add JUnit Tests

NewtonKoh committed:
Create a JAR File

NewtonKoh committed:
JavaDoc

NewtonKoh committed:
Follow the Coding Standard

NewtonKoh committed:
Find

NewtonKoh committed:
Use CheckStyle

NewtonKoh committed:
GUI

NewtonKoh committed:
Use Varargs (Not applicable)

NewtonKoh committed:
Implement ChatGPT to answer trivia

There are several methods that allow for trivia handling.

Using ChatGPT's API improves trivia handling by a great amount.

As a step towards better handling trivia, let's integrate ChatGPT's API into our handling of user messages. Doing so will make handling trivia easier.

NewtonKoh committed:
Add better user interface

NewtonKoh committed:
Use assertions to catch undefined behaviour

NewtonKoh committed:
Improve the quality of the code

NewtonKoh committed:
Add error handling for empty TaskList

NewtonKoh committed:
Merge pull request #2 from NewtonKoh/branch-A-MoreErrorHandling

Add error handling for empty TaskList

NewtonKoh committed:
Add screenshot of GUI

NewtonKoh committed:
Update GFMD style user guide

NewtonKoh committed:
Update README.md in root directory

NewtonKoh committed:
Update user guide details

NewtonKoh committed:
Update Ui.png and improve GUI
